test/foundry/erc721/peripheral-erc721-pool/bulk_withdraw/bulkWithdraw.t.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.4 <0.9.0;
+
+import { IPeripheralERC721Pool } from "contracts/ERC-721/IPeripheralERC721Pool.sol";
+import { PeripheralERC721Pool_Test } from "../PeripheralERC721Pool.t.sol";
+
+contract BulkWithdraw_Test is PeripheralERC721Pool_Test {
+    /// @dev it should revert.
+    function test_RevertWhen_IdsIsEmpty() external {
+        uint256[] memory ids;
+        vm.expectRevert(IPeripheralERC721Pool.PeripheralERC721Pool__InsufficientIn.selector);
+        peripheralERC721Pool.bulkWithdraw(erc721Pool, ids);
+    }
+
+    /// @dev it should revert.
+    function test_RevertWhen_AllowanceNotEnough(uint256[] memory ids) external {
+        vm.assume(ids.length > 0);
+        // For this test do not approve PeripheralERC721Pool to transfer pool token from the caller.
+        vm.expectRevert(IPeripheralERC721Pool.PeripheralERC721Pool__UnapprovedOperator.selector);
+        peripheralERC721Pool.bulkWithdraw(erc721Pool, ids);
+    }
+
+    /// @dev It should revert.
+    function test_RevertWhen_IdNotAvailableInPool() external {
+        // Set up the test environment with the given number of available NFTs.
+
+        // Mint NFTs to erc721Pool.
+        uint256[] memory availableIds = new uint256[](5);
+        for (uint256 i = 0; i < 5; i++) {
+            availableIds[i] = i;
+        }
+        mintNft(address(erc721Pool), availableIds);
+
+        // Mint erc721Pool tokens so that we have enough to withdraw.
+        erc721Pool.__godMode_mint(users.alice, 5 * 10**18);
+        erc721Pool.__godMode_setHoldings(availableIds);
+
+        // Request 3 NFTs out which last one is not available.
+        uint256[] memory ids = new uint256[](3);
+        ids[0] = 1;
+        ids[1] = 2;
+        ids[2] = 12;
+
+        erc721Pool.approve(address(peripheralERC721Pool), ids.length * 10**18);
+        vm.expectRevert();
+        peripheralERC721Pool.bulkWithdraw(erc721Pool, ids);
+        uint256 erc721PoolTokenBalance = erc721Pool.balanceOf(users.alice);
+        uint256 erc721AssetBalance = nft.balanceOf(address(erc721Pool));
+        assertEq(erc721PoolTokenBalance, 5 * 10**18, "caller pool tokens balace should remain same");
```

Comment:
Slight typo `balace` shoud probably be `balance`

Reply:
Thank @hroussille for pointing out the typo. It is fixed now.